pandas/io/clipboards.py

```diff
@@ -58,7 +58,9 @@ def read_clipboard(sep=r"\s+", **kwargs):  # pragma: no cover
     # 0  1  2
     # 1  3  4
 
-    counts = {x.lstrip().count("\t") for x in lines}
+    counts = {
+        x.count("\t") if i > 0 else x.lstrip().count("\t") for i, x in enumerate(lines)
+    }
```

Comment:
Does this solve the issue when columns are a MultiIndex? Would using `.lstrip(" ")` be better?

Reply:
You are right it doesn't work well with MultiIndex, in general it doesnt work when you have an Index at all

The problem happens when the separator is "\s+" and there are nulls in the columns/index

There was this comment in the code and i didn't want to break the behavior, so to keep it as it was, if there is a index the separator remains as `\s+`
```
# Need to remove leading white space, since read_csv
# accepts:
#    a  b
# 0  1  2
# 1  3  4
```

But that means the values still end up in the wrong column for those cases

 `.lstrip(" ")` or just removing the .lstrip completely as in:
```python
counts = {x.count("\t") for x in lines}
```
would correctly infer excel and set `sep="\t"`, but then it does not set the indexes by default, and instead they show up as `Unnamed: 0` columns, I still think that's preferable to reading the values in the wrong column though

but for example it has the consequence of turning this:
```
		a	b
A	0	1	2
A	1	3	4
A	3	5	6

```
```python
In [3]: pandas.read_clipboard()
Out[3]: 
     a  b
A 0  1  2
  1  3  4
  3  5  6
```

into this:
```python
In [8]: pandas.read_clipboard()
Out[8]: 
  Unnamed: 0  Unnamed: 1  a  b
0          A           0  1  2
1          A           1  3  4
2          A           3  5  6
```

on the other hand, if there are nulls, or merged cells as the index, the current behavior is going to give you the wrong results
```
		a	b
A	0	1	2
A	1		4
A	3	5	6

```
```python
In [11]: pandas.read_clipboard()
Out[11]: 
     a    b
A 0  1  2.0
  1  4  NaN
  3  5  6.0
```

I really dont know what's better considering this is an interactive function

Reply:
> You are right it doesn't work well with MultiIndex, in general it doesnt work when you have an Index at all

Is this remark about this PR, or is it about the current state even in master?

> I really dont know what's better considering this is an interactive function

In general, my opinion is that it is better to me to have stability. In other words, don't fix one bug at the expense of causing another. Gets tricky when one is viewed as serious and the other as minor.

Reply:
> Is this remark about this PR, or is it about the current state even in master?

Both, when there are Indexes my changed did not change the behavior, but also did not fix the problem with the columns shifting left

I've just made some changes that I think should make it work